app/routes/assignment/assignment.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { IRouterContext } from 'koa-router';
+import { AssignmentStatus } from '../../models';
+import { INTERNAL_SERVER_ERROR, OK } from 'http-status-codes';
+import { setResponse } from '../utils';
+
+const createPatchAssignmentRoute = async (ctx: any, next: any) => {
+    ctx.body.status = AssignmentStatus.Checked;
```

Comment:
You shouldn't create your own functions and test them. You should test `createPatchAssignmentRoute ` from `app/routes/assignment/assignment.ts`. I mean that in your test cases you should call `createPatchAssignmentRoute `, but before you should mock AssignmentModel.
For Example:
```js
import { createPatchAssignmentRoute } from './assignment';
import { AssignmentModel} from '../../models';

jest.mock('../../models', () => ({
    AssignmentModel: { findByIdAndUpdate : jest.fn().mockReturnValue(someObjectHere) },
}));

describe('app/routes/assignment/assignment.ts', () => {
    it('returns 200 if returns data', async () => {
        const ctx = createTestContext();
        await createPatchAssignmentRoute(ctx, findByIdAndUpdate);
        expect(AssignmentModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
        expect(ctx).toMatchSnapshot();
    });
```

Reply:
The same for all tests